dips/dip-183.md

```diff
@@ -0,0 +1,100 @@
+---
+dip: 183
+title: Sender Initiated Reference ID Exchange
+authors: Sunmi Lee (@sunmilee), David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 06/08/2021
+last_updated: 06/08/2021
+issue: https://github.com/diem/dip/issues/183
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Summary
+
+This DIP describes a protocol standard for sender initialized pre-flight exchange using reference ID to form payment transactions.
+
+Reference ID is used as a unique identifier for a payment transaction between two parties across two VASPs without revealing any information about those parties on-chain.
```

Comment:
IIUC, each VASP will need to maintain every unique (reference id, sending VASP, receiving VASP) identifier to ensure avoiding duplicate reference ids?

Can VASP A initiate a reference ID R to VASP B
and VASP B also legally initiate a reference ID R to VASP A or is it unique per pair?

Reply:
Unique per pair, so between VASP A and B, if a reference ID R was used in a previous transaction, regardless of which one initiated it, R cannot be reused for any future transactions between A and B

Reply:
Thanks for the clarification.  I think it would be good to add something like your statement into the DIP to make this clear.

We should consider from an implementor perspective how feasible this is to implement.  I guess it is expected that the implementor would need to maintain an index of (canonical(VASP1, VASP2), reference ID) and rebuild it from the event history if they lost the index.

This might be simpler if, say, there was just a monotonically increasing reference ID number so one could just look at the last number between canonical(VASP1, VASP, last reference ID number) event on-chain rather than keeping an index?

Reply:
I think it depends on how the VASP decides to implement it. They can always enforce uniqueness across all reference IDs instead of pair, which is what I do in my miniwallet diem ID example.